lib/src/kbasesearchengine/search/ElasticIndexingStorage.java

```diff
@@ -1077,22 +1087,39 @@ public void unpublishObjectsExternally(Set<GUID> guids, int accessGroupId) throw
         for (Map<String, Object> hit : hitList) {
             @SuppressWarnings("unchecked")
             Map<String, Object> obj = (Map<String, Object>) hit.get("_source");
-            ObjectData item = buildObjectData(obj, pp.objectInfo, pp.objectKeys, 
-                    pp.objectData, pp.objectDataIncludes);
+            Map<String, ArrayList> highlightRes = (Map<String, ArrayList>) hit.get("highlight");
+            ObjectData item = buildObjectData(obj, editHighlightedKeys(highlightRes), pp);
             ret.add(item);
         }
         return ret;
     }
 
+    private Map<String, ArrayList> editHighlightedKeys(Map<String, ArrayList> highlightRes){
+        if(highlightRes == null){
+            return null;
+        }
+        Map<String, ArrayList> highlightResEdited = new HashMap<>();
+        for(String key : highlightRes.keySet()){
+            String newKey = key;
+            if(key.startsWith("key.")){
```

Comment:
Is there functionality like this anywhere else in the code? I seem to remember something similar and if so we should consolidate into a method.

Reply:
I dont think so? there is functionality that that add the "key." to fields but not one to strip the key

Reply:
There's something vaguely similar in buildObjectData